packages/react-aria/src/hooks/useAriaButton.ts

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { ObjectShorthandProps, useEventCallback } from '@fluentui/react-utilities';
+
+export function useARIAButton(shorthand: ObjectShorthandProps<React.ButtonHTMLAttributes<HTMLElement>>) {
+  if (!shorthand.hasOwnProperty('as')) {
+    shorthand.as = 'button';
+  }
+  if (!shorthand.hasOwnProperty('children')) {
+    shorthand.children = null;
+  }
+
+  const { onClick, onKeyDown, onKeyUp, as, disabled } = shorthand;
+
+  const onClickHandler = useEventCallback((ev: React.MouseEvent<HTMLElement>) => {
```

Comment:
well if a rerender happens you get a new closer, internally `useEventCallback` just always uses the new closure and assigns it to a ref to avoid unnecessary updates